internal/provider/providerfunction/normalise_resource_id.go

```diff
@@ -0,0 +1,48 @@
+package providerfunction
```

Comment:
provider seems redundant here? should we just call this functions?
```suggestion
package functions
```

Reply:
I think that `function` is an overloaded term in the provider (given the number of resources using it also) - So it seemed appropriate to be explicit/verbose here that this package is exclusively for Provider Functions to avoid `helper` functions being added here in future? I'm not particularly set on that though, but felt it worth explaining my thinking. WDYT?